### DISCUSSION TITLE
CORS blocked with deleted frontend and nextjs app instead

### DISCUSSION THREAD

**Discussion by JimVincentW** [fastapi/full-stack-fastapi-template#1204] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
useEffect(() => {
    if (country) {
      const fetchLeaderInfo = async () => {
        try {
          const response = await axios.get(`http://dev.opp.com/leader-info/${country}`);  // doesn't work with serverip or localhost either
          setLeaderInfo(response.data);
        } catch (error) {
          console.error('Error fetching leader info:', error);
          console.error('Error details:', error.toJSON());
        }
      };
  
      fetchLeaderInfo();

    }
  }, [country]);
```

### Description

Hello Community,

I was not comfortable with chakra UI so I removed the frontend like instructued in the frontend/readme.md.

I swapped it out for a nextjs app. I am running the backend container in docker and the nextjs app locally (both on my server).

My frontend api never reaches the fastapi backend.
I can not resolve this error:
Access to XMLHttpRequest at 'http://dev.opp.com/leader-info/Cameroon' from origin 'http://dev.opp.com:3001' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

It's added to my .env file but I keep running into this error. 
From my .env file:
DOMAIN=dev.opp.com

BACKEND_CORS_ORIGINS="http://localhost,http://localhost:5173,https://localhost,https://localhost:5173,http://localhost.tiangolo.com,https://dev.opp.com,http://dev.opp.com:5173,http://localhost:8088,https://staging.opp.com,http://staging.opp.com:5173,https://app.open-politics.org,http://app.open-politics.org:5173,http://dev.opp.com:3001,http://localhost:3001,http://localhost:3000,http://dev.opp.com:3001,http://136.243.80.175,https://136.243.80.175"

Can somebody help me out or point me in the right direction?

Logs:
backend-1   | Checking for script in /app/prestart.sh
backend-1   | Running script /app/prestart.sh
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Provider event received {Status:start ID:e0fc6a5ec6792b63f00a66c9728e24a8709b9f20639163cf9ab4cf228d32fb59 From:backend:latest Type:container Action:start Actor:{ID:e0fc6a5ec6792b63f00a66c9728e24a8709b9f20639163cf9ab4cf228d32fb59 Attributes:map[com.docker.compose.config-hash:abe02d92aa8f13f07ff40a1055f6ce0da0788cb4162ef0ae48d0ae849754a6a0 com.docker.compose.container-number:1 com.docker.compose.depends_on:db:service_started:false com.docker.compose.image:sha256:81d9cc6bd068da25dac5498bddb277ff7ac4cbd9914332078087639d2278c60a com.docker.compose.oneoff:False com.docker.compose.project:open-politics com.docker.compose.project.config_files:/root/projects/open-politics/docker-compose.yml,/root/projects/open-politics/docker-compose.override.yml com.docker.compose.project.working_dir:/root/projects/open-politics com.docker.compose.service:backend com.docker.compose.version:2.27.0 image:backend:latest maintainer:Sebastian Ramirez <tiangolo@gmail.com> name:open-politics-backend-1 traefik.constraint-label:traefik-public traefik.docker.network:traefik-public traefik.enable:true traefik.http.routers.open-politics-backend-http.entrypoints:http traefik.http.routers.open-politics-backend-http.middlewares:https-redirect,open-politics-www-redirect traefik.http.routers.open-politics-backend-http.rule:Host(`dev.opp.com`, `www.dev.opp.com`) && PathPrefix(`/api`, `/docs`, `/redoc`) traefik.http.routers.open-politics-backend-https.entrypoints:https traefik.http.routers.open-politics-backend-https.middlewares:open-politics-www-redirect traefik.http.routers.open-politics-backend-https.rule:Host(`dev.opp.com`, `www.dev.opp.com`) && PathPrefix(`/api`, `/docs`, `/redoc`) traefik.http.routers.open-politics-backend-https.tls:true traefik.http.routers.open-politics-backend-https.tls.certresolver:le traefik.http.services.open-politics-backend.loadbalancer.server.port:80]} Scope:local Time:1715832492 TimeNano:1715832492349388679}" providerName=docker
proxy-1     | time="2024-05-16T04:08:12Z" level=warning msg="Could not find network named 'traefik-public' for container '/open-politics-backend-1'! Maybe you're missing the project's prefix in the label? Defaulting to first available network." providerName=docker container=backend-open-politics-e0fc6a5ec6792b63f00a66c9728e24a8709b9f20639163cf9ab4cf228d32fb59 serviceName=open-politics-backend
proxy-1     | time="2024-05-16T04:08:12Z" level=warning msg="Could not find network named 'traefik-public' for container '/open-politics-adminer-1'! Maybe you're missing the project's prefix in the label? Defaulting to first available network." serviceName=open-politics-adminer providerName=docker container=adminer-open-politics-6f5a1e8bf2cb2bcbfc5ce1c966f5661999e56cd671ab3aaa36eaec6027c55d8b
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=db-open-politics-770cb4c1e15331ece3532d256ca1b975d03a26e9298fe0ef86f0d21599caeadd
proxy-1     | time="2024-05-16T04:08:12Z" level=warning msg="Could not find network named 'traefik-public' for container '/open-politics-frontend-1'! Maybe you're missing the project's prefix in the label? Defaulting to first available network." serviceName=open-politics-frontend providerName=docker container=frontend-open-politics-eafaa2a09ecc13b20b1b08dc847e6e8d3969c8055acc7f19d2df4ff9e997c233
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" container=geo-service-ssare-b4d69d8b010530ec7f69fe018ad49fe14fd5973bbf9ebdf0f139424151d2a4ec providerName=docker
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" container=qdrant-service-ssare-30a664c34a31c55d392c55f65c249bdcde23915dd11bbef1d6fe85166ee19124 providerName=docker
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=postgres-service-ssare-b669ce4bc4f1cdc4de75ae1f315363728744d6e2f0d4c4ba57190b447eea82b5
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=entity-service-ssare-7188a59640b8d59fb4174a45e563866e7f45ebdea78d47e619118359db1b2d71
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=prefect-agent-ssare-d985610675093a6db7f0499524db6a94db076658158cf67dc8edba0f800d260c
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=nlp-service-ssare-8510f3c5666b24a94120ce04c826cc6b5812e89968b863c644db3aac303737ed
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=scraper-service-ssare-c5d6031f014de99ba3b370200fac5eeb1a6efec6f2d06221ecf3e5cfdaaee416
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=prefect-server-ssare-8c18cb6189cc6f8af95438115d05c662799f0dc679fd0bf876306521cc2b53f9
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=postgres-db-ssare-2bb0bb4499c161cdf9e8df89dc035792419b4f1850ca4f8da2bb600bdff4a1d6
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" container=pelias-placeholder-ssare-2b5d01eb1f7851364afdd7651687113b0f4acdfbe64d92556aa53566cef9fd56 providerName=docker
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=redis-ssare-58cfba6b00b47160f2b6e321d7ad63e0e8b50108b593e0bd7e3e8a8e755ae2d8
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" providerName=docker container=prefect-database-ssare-cc95fac705648faeb3450f1df9fa456194986f7bc35d248255d8799598fdf0d4
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Filtering disabled container" container=qdrant-storage-ssare-67ab679997852f945ec1fbf21ce4d150dccbef76c1e9e8ec6780b73fc55924df providerName=docker
proxy-1     | time="2024-05-16T04:08:12Z" level=debug msg="Configuration received from provider docker: {\"http\":{\"routers\":{\"open-politics-adminer-http\":{\"entryPoints\":[\"http\"],\"middlewares\":[\"https-redirect\"],\"service\":\"open-politics-adminer\",\"rule\":\"Host(`adminer.dev.opp.com`)\"},\"open-politics-adminer-https\":{\"entryPoints\":[\"https\"],\"service\":\"open-politics-adminer\",\"rule\":\"Host(`adminer.dev.opp.com`)\",\"tls\":{\"certResolver\":\"le\"}},\"open-politics-backend-http\":{\"entryPoints\":[\"http\"],\"middlewares\":[\"https-redirect\",\"open-politics-www-redirect\"],\"service\":\"open-politics-backend\",\"rule\":\"Host(`dev.opp.com`, `www.dev.opp.com`) \\u0026\\u0026 PathPrefix(`/api`, `/docs`, `/redoc`)\"},\"open-politics-backend-https\":{\"entryPoints\":[\"https\"],\"middlewares\":[\"open-politics-www-redirect\"],\"service\":\"open-politics-backend\",\"rule\":\"Host(`dev.opp.com`, `www.dev.opp.com`) \\u0026\\u0026 PathPrefix(`/api`, `/docs`, `/redoc`)\",\"tls\":{\"certResolver\":\"le\"}},\"open-politics-frontend-http\":{\"entryPoints\":[\"http\"],\"middlewares\":[\"https-redirect\",\"open-politics-www-redirect\"],\"service\":\"open-politics-frontend\",\"rule\":\"Host(`dev.opp.com`, `www.dev.opp.com`)\"},\"open-politics-frontend-https\":{\"entryPoints\":[\"https\"],\"middlewares\":[\"open-politics-www-redirect\"],\"service\":\"open-politics-frontend\",\"rule\":\"Host(`dev.opp.com`, `www.dev.opp.com`)\",\"tls\":{\"certResolver\":\"le\"}},\"proxy-open-politics\":{\"service\":\"proxy-open-politics\",\"rule\":\"Host(`proxy-open-politics`)\"}},\"services\":{\"open-politics-adminer\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.176.5:8080\"}],\"passHostHeader\":true}},\"open-politics-backend\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.176.6:80\"}],\"passHostHeader\":true}},\"open-politics-frontend\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.176.4:3000\"}],\"passHostHeader\":true}},\"proxy-open-politics\":{\"loadBalancer\":{\"servers\":[{\"url\":\"http://192.168.176.2:80\"}],\"passHostHeader\":true}}},\"middlewares\":{\"https-redirect\":{\"contentType\":{}},\"open-politics-www-redirect\":{\"redirectRegex\":{\"regex\":\"^http(s)?://www.(dev.opp.com)/(.*)\",\"replacement\":\"http${1}://dev.opp.com/${3}\"}}}},\"tcp\":{},\"udp\":{}}" providerName=docker
backend-1   | INFO:__main__:Initializing service
backend-1   | INFO:__main__:Starting call to '__main__.init', this is the 1st time calling it.
backend-1   | INFO:__main__:Service finished initializing
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="No entryPoint defined for this router, using the default one(s) instead: [http https]" routerName=proxy-open-politics
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" serviceName=open-politics-backend middlewareName=pipelining middlewareType=Pipelining entryPointName=http routerName=open-politics-backend-http@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" routerName=open-politics-backend-http@docker serviceName=open-politics-backend entryPointName=http
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.6:80" serverName=0 entryPointName=http routerName=open-politics-backend-http@docker serviceName=open-politics-backend
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-backend" routerName=open-politics-backend-http@docker middlewareName=tracing middlewareType=TracingForwarder entryPointName=http
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=http routerName=open-politics-backend-http@docker middlewareName=open-politics-www-redirect@docker middlewareType=RedirectRegex
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Setting up redirection from ^http(s)?://www.(dev.opp.com)/(.*) to http${1}://dev.opp.com/${3}" entryPointName=http routerName=open-politics-backend-http@docker middlewareName=open-politics-www-redirect@docker middlewareType=RedirectRegex
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" entryPointName=http routerName=open-politics-backend-http@docker middlewareName=open-politics-www-redirect@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareName=pipelining middlewareType=Pipelining routerName=open-politics-frontend-http@docker entryPointName=http serviceName=open-politics-frontend
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" routerName=open-politics-frontend-http@docker entryPointName=http serviceName=open-politics-frontend
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.4:3000" serverName=0 entryPointName=http serviceName=open-politics-frontend routerName=open-politics-frontend-http@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-frontend" middlewareName=tracing middlewareType=TracingForwarder entryPointName=http routerName=open-politics-frontend-http@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=http routerName=open-politics-frontend-http@docker middlewareType=RedirectRegex middlewareName=open-politics-www-redirect@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Setting up redirection from ^http(s)?://www.(dev.opp.com)/(.*) to http${1}://dev.opp.com/${3}" entryPointName=http routerName=open-politics-frontend-http@docker middlewareType=RedirectRegex middlewareName=open-politics-www-redirect@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" routerName=open-politics-frontend-http@docker middlewareName=open-politics-www-redirect@docker entryPointName=http
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" serviceName=proxy-open-politics middlewareType=Pipelining middlewareName=pipelining entryPointName=http routerName=proxy-open-politics@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" routerName=proxy-open-politics@docker serviceName=proxy-open-politics entryPointName=http
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.2:80" routerName=proxy-open-politics@docker serviceName=proxy-open-politics entryPointName=http serverName=0
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware proxy-open-politics" middlewareType=TracingForwarder entryPointName=http routerName=proxy-open-politics@docker middlewareName=tracing
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareName=pipelining middlewareType=Pipelining serviceName=open-politics-adminer entryPointName=http routerName=open-politics-adminer-http@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" entryPointName=http routerName=open-politics-adminer-http@docker serviceName=open-politics-adminer
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.5:8080" serverName=0 entryPointName=http routerName=open-politics-adminer-http@docker serviceName=open-politics-adminer
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-adminer" middlewareName=tracing entryPointName=http routerName=open-politics-adminer-http@docker middlewareType=TracingForwarder
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=http middlewareName=traefik-internal-recovery middlewareType=Recovery
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware dashboard@internal" routerName=dashboard@internal entryPointName=traefik middlewareName=tracing middlewareType=TracingForwarder
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_stripprefix@internal middlewareType=StripPrefix
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_stripprefix@internal
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareType=RedirectRegex entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_redirect@internal
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Setting up redirection from ^(http:\\/\\/(\\[[\\w:.]+\\]|[\\w\\._-]+)(:\\d+)?)\\/$ to ${1}/dashboard/" middlewareType=RedirectRegex entryPointName=traefik routerName=dashboard@internal middlewareName=dashboard_redirect@internal
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" routerName=dashboard@internal middlewareName=dashboard_redirect@internal entryPointName=traefik
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware api@internal" routerName=api@internal middlewareName=tracing middlewareType=TracingForwarder entryPointName=traefik
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=traefik middlewareName=traefik-internal-recovery middlewareType=Recovery
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareType=Recovery entryPointName=https middlewareName=traefik-internal-recovery
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareType=Pipelining entryPointName=https routerName=open-politics-adminer-https@docker serviceName=open-politics-adminer middlewareName=pipelining
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" entryPointName=https routerName=open-politics-adminer-https@docker serviceName=open-politics-adminer
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.5:8080" routerName=open-politics-adminer-https@docker serviceName=open-politics-adminer serverName=0 entryPointName=https
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-adminer" routerName=open-politics-adminer-https@docker middlewareName=tracing middlewareType=TracingForwarder entryPointName=https
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" serviceName=open-politics-backend middlewareName=pipelining middlewareType=Pipelining entryPointName=https routerName=open-politics-backend-https@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" serviceName=open-politics-backend entryPointName=https routerName=open-politics-backend-https@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.6:80" entryPointName=https routerName=open-politics-backend-https@docker serverName=0 serviceName=open-politics-backend
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-backend" middlewareName=tracing middlewareType=TracingForwarder entryPointName=https routerName=open-politics-backend-https@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" routerName=open-politics-backend-https@docker middlewareName=open-politics-www-redirect@docker middlewareType=RedirectRegex entryPointName=https
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Setting up redirection from ^http(s)?://www.(dev.opp.com)/(.*) to http${1}://dev.opp.com/${3}" routerName=open-politics-backend-https@docker middlewareName=open-politics-www-redirect@docker middlewareType=RedirectRegex entryPointName=https
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" entryPointName=https middlewareName=open-politics-www-redirect@docker routerName=open-politics-backend-https@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=https routerName=open-politics-frontend-https@docker serviceName=open-politics-frontend middlewareName=pipelining middlewareType=Pipelining
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating load-balancer" serviceName=open-politics-frontend entryPointName=https routerName=open-politics-frontend-https@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating server 0 http://192.168.176.4:3000" serverName=0 entryPointName=https routerName=open-politics-frontend-https@docker serviceName=open-politics-frontend
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Added outgoing tracing middleware open-politics-frontend" entryPointName=https routerName=open-politics-frontend-https@docker middlewareName=tracing middlewareType=TracingForwarder
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" middlewareType=RedirectRegex entryPointName=https routerName=open-politics-frontend-https@docker middlewareName=open-politics-www-redirect@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Setting up redirection from ^http(s)?://www.(dev.opp.com)/(.*) to http${1}://dev.opp.com/${3}" entryPointName=https routerName=open-politics-frontend-https@docker middlewareName=open-politics-www-redirect@docker middlewareType=RedirectRegex
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Adding tracing to middleware" entryPointName=https routerName=open-politics-frontend-https@docker middlewareName=open-politics-www-redirect@docker
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="Creating middleware" entryPointName=https middlewareName=traefik-internal-recovery middlewareType=Recovery
proxy-1     | time="2024-05-16T04:08:13Z" level=debug msg="No default certificate, generating one"
proxy-1     | time="2024-05-16T04:08:14Z" level=debug msg="Adding route for dev.opp.com with TLS options default" entryPointName=https
proxy-1     | time="2024-05-16T04:08:14Z" level=debug msg="Adding route for www.dev.opp.com with TLS options default" entryPointName=https
proxy-1     | time="2024-05-16T04:08:14Z" level=debug msg="Adding route for adminer.dev.opp.com with TLS options default" entryPointName=https
proxy-1     | time="2024-05-16T04:08:14Z" level=error msg="the router open-politics-backend-https@docker uses a non-existent resolver: le"
proxy-1     | time="2024-05-16T04:08:14Z" level=error msg="the router open-politics-adminer-https@docker uses a non-existent resolver: le"
proxy-1     | time="2024-05-16T04:08:14Z" level=error msg="the router open-politics-frontend-https@docker uses a non-existent resolver: le"
backend-1   | INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
backend-1   | INFO  [alembic.runtime.migration] Will assume transactional DDL.
backend-1   | INFO:__main__:Creating initial data
backend-1   | INFO:__main__:Initial data created
backend-1   | INFO:     Will watch for changes in these directories: ['/app']
backend-1   | INFO:     Uvicorn running on http://0.0.0.0:80 (Press CTRL+C to quit)
backend-1   | INFO:     Started reloader process [1] using WatchFiles
backend-1   | INFO:     Started server process [11]
backend-1   | INFO:     Waiting for application startup.
backend-1   | INFO:     Application startup complete.
adminer-1   | [Thu May 16 04:08:33 2024] [::ffff:192.168.176.1]:36974 Accepted
proxy-1     | time="2024-05-16T04:08:50Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/leader-info/Cameroon\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\",\"RawFragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"application/json, text/plain, */*\"],\"Accept-Encoding\":[\"gzip, deflate\"],\"Accept-Language\":[\"en-US,en;q=0.9,de;q=0.8\"],\"Connection\":[\"keep-alive\"],\"Dnt\":[\"1\"],\"Origin\":[\"http://dev.opp.com:3001\"],\"Referer\":[\"http://dev.opp.com:3001/\"],\"User-Agent\":[\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36\"],\"X-Forwarded-Host\":[\"dev.opp.com\"],\"X-Forwarded-Port\":[\"80\"],\"X-Forwarded-Proto\":[\"http\"],\"X-Forwarded-Server\":[\"ee0cab113789\"],\"X-Real-Ip\":[\"92.193.193.129\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"dev.opp.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"92.193.193.129:52725\",\"RequestURI\":\"/leader-info/Cameroon\",\"TLS\":null}"
proxy-1     | time="2024-05-16T04:08:50Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/leader-info/Cameroon\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\",\"RawFragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"application/json, text/plain, */*\"],\"Accept-Encoding\":[\"gzip, deflate\"],\"Accept-Language\":[\"en-US,en;q=0.9,de;q=0.8\"],\"Connection\":[\"keep-alive\"],\"Dnt\":[\"1\"],\"Origin\":[\"http://dev.opp.com:3001\"],\"Referer\":[\"http://dev.opp.com:3001/\"],\"User-Agent\":[\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36\"],\"X-Forwarded-Host\":[\"dev.opp.com\"],\"X-Forwarded-Port\":[\"80\"],\"X-Forwarded-Proto\":[\"http\"],\"X-Forwarded-Server\":[\"ee0cab113789\"],\"X-Real-Ip\":[\"92.193.193.129\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"dev.opp.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"92.193.193.129:52725\",\"RequestURI\":\"/leader-info/Cameroon\",\"TLS\":null}" ForwardURL="http://192.168.176.4:3000"
proxy-1     | time="2024-05-16T04:08:50Z" level=debug msg="'502 Bad Gateway' caused by: dial tcp 192.168.176.4:3000: connect: connection refused"
proxy-1     | time="2024-05-16T04:08:50Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/leader-info/Cameroon\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\",\"RawFragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"application/json, text/plain, */*\"],\"Accept-Encoding\":[\"gzip, deflate\"],\"Accept-Language\":[\"en-US,en;q=0.9,de;q=0.8\"],\"Connection\":[\"keep-alive\"],\"Dnt\":[\"1\"],\"Origin\":[\"http://dev.opp.com:3001\"],\"Referer\":[\"http://dev.opp.com:3001/\"],\"User-Agent\":[\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36\"],\"X-Forwarded-Host\":[\"dev.opp.com\"],\"X-Forwarded-Port\":[\"80\"],\"X-Forwarded-Proto\":[\"http\"],\"X-Forwarded-Server\":[\"ee0cab113789\"],\"X-Real-Ip\":[\"92.193.193.129\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"dev.opp.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"92.193.193.129:52725\",\"RequestURI\":\"/leader-info/Cameroon\",\"TLS\":null}"
proxy-1     | 92.193.193.129 - - [16/May/2024:04:08:50 +0000] "GET /leader-info/Cameroon HTTP/1.1" 502 11 "-" "-" 1 "open-politics-frontend-http@docker" "http://192.168.176.4:3000" 2ms
adminer-1   | [Thu May 16 04:09:02 2024] [::ffff:89.190.156.248]:61001 Accepted
adminer-1   | [Thu May 16 04:09:04 2024] [::ffff:89.190.156.248]:61001 [404]: (null) /cgi-bin/nas_sharing.cgi - No such file or directory
adminer-1   | [Thu May 16 04:09:04 2024] [::ffff:89.190.156.248]:61001 Closing
adminer-1   | [Thu May 16 04:09:35 2024] [::ffff:192.168.176.1]:36974 Closed without sending a request; it was probably just an unused speculative preconnection
adminer-1   | [Thu May 16 04:09:35 2024] [::ffff:192.168.176.1]:36974 Closing
proxy-1     | 92.193.193.129 - - [16/May/2024:04:11:03 +0000] "GET /api/overview HTTP/1.1" 200 435 "-" "-" 2 "api@internal" "-" 0ms
proxy-1     | 92.193.193.129 - - [16/May/2024:04:11:06 +0000] "GET /api/overview HTTP/1.1" 200 435 "-" "-" 3 "api@internal" "-" 0ms

### Operating System

Linux

### Operating System Details

Ubuntu 22.04 Server (Remote)/ M1 Macbook (local)

### Python Version

3.10.2

### Additional Context

Thankful for any help!!

**Comment by tiangolo** [comment, marked as answer]
Hello there, you can start by reading the docs for CORS here: https://fastapi.tiangolo.com/tutorial/cors/

You can also read the docs about CORS in the Mozilla docs.

You might want to add some logs to your backend to show the currently used CORS origins before the setup. It could be that you haven't rebuilt and restated your Docker image after changing CORS or something similar. Printing the CORS origins in your code would help you debug if the problem is before or after that point.
